common/src/main/java/org/conscrypt/OpenSSLCipher.java

```diff
@@ -1167,6 +1167,22 @@ void engineInitInternal(byte[] encodedKey, AlgorithmParameterSpec params,
             reset();
         }
 
+        @Override
+        protected int engineDoFinal(byte[] input, int inputOffset, int inputLen, byte[] output,
+                int outputOffset) throws ShortBufferException, IllegalBlockSizeException,
+                BadPaddingException {
+            // Because the EVP_AEAD updateInternal processes input but doesn't create any output
+            // (and thus can't check the output buffer), we need to add this check before the
+            // superclass' processing to ensure that updateInternal is never called if the
+            // output buffer isn't large enough.
+            if (output != null) {
+                if (getOutputSizeForFinal(inputLen) > output.length - outputOffset) {
+                    throw new ShortBufferException("Insufficient output space");
```

Comment:
So I see where you're throwing it from Java ... but where is the new native exception being used?

Reply:
The specific place I know of that this will trigger is line 3539 in native_crypto.cc, which is called when an EVP_AEAD_CTX_seal or EVP_AEAD_CTX_open call returns failure.  The EVP_AEAD calls check the bounds on the output and return CIPHER_R_BUFFER_TOO_SMALL if there isn't enough output space.

Reply:
Ok, so I think what you're saying is that the previous java exception handling will properly handle `ShortBufferException` (i.e. already catching a super class)? I just wasn't seeing where in the java code you were handling the new `ShortBufferException` thrown by c.

Reply:
Oh, I understand the question now.  We're not handling it anywhere.  The top-level Cipher operations like doFinal() declare that they throw ShortBufferException, we just weren't actually throwing it when users passed us a short buffer in some situations.